staging/src/k8s.io/apiserver/pkg/server/options/authentication.go

```diff
@@ -97,6 +101,9 @@ func (s *RequestHeaderAuthenticationOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.StringSliceVar(&s.UsernameHeaders, "requestheader-username-headers", s.UsernameHeaders, ""+
 		"List of request headers to inspect for usernames. X-Remote-User is common.")
 
+	fs.StringSliceVar(&s.UIDHeaders, "requestheader-uid-headers", s.UIDHeaders, ""+
+		"List of request headers to inspect for UIDs. X-Remote-UID is suggested.")
```

Comment:
Update all `UID` in header keys to be `Uid` to follow how Go canonicalizes these headers.

```suggestion
		"List of request headers to inspect for UIDs. X-Remote-Uid is suggested.")
```